sqldb/sqlc/migrations/000002_amp_invoices.up.sql

```diff
@@ -10,26 +10,35 @@ CREATE TABLE IF NOT EXISTS amp_invoice_payments (
     -- Timestamp of when the first htlc for this payment was accepted.
     created_at TIMESTAMP NOT NULL,
 
-    -- If settled, the invoice payment related to this set id.
-    settled_index INTEGER REFERENCES invoice_payments(id),
+    -- When the invoice was settled.
+    settled_at TIMESTAMP,
 
-    -- The invoice id this set id is related to.
-    invoice_id INTEGER NOT NULL REFERENCES invoices(id)
+    -- If settled, the index is set to the current_value+1 of the settled_index
+    -- seuqence in the invoice_sequences table. If not settled,then it is set
+    -- to NULL.
+    settled_index BIGINT,
+
+    -- The id of the parent invoice this AMP invoice belongs to.
+    invoice_id BIGINT NOT NULL REFERENCES invoices(id) ON DELETE CASCADE,
+
+    -- A unique constraint on the set_id and invoice_id is needed to ensure that
+    -- we don't have two AMP invoices for the same invoice.
```

Comment:
Where will the individual payments to the invoice be tracked then? For a given invoice ID, you _can_ have many other `set_id` values, as those are the new unique identifier (new ID for each settled payment).